src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -195,13 +217,17 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
                                                                        options.ortoolsSolver);
     auto legacyOrtoolsFiller = std::make_unique<LegacyFiller>(&Probleme);
     std::vector<LinearProblemFiller*> fillersCollection = {legacyOrtoolsFiller.get()};
-    LinearProblemData LP_Data;
-    FillContext fillCtx(0, 167);
+    std::vector<std::unique_ptr<ComponentFiller>> componentFillers;
+
+    fillModelerComponents(componentFillers, fillersCollection, problemeHebdo->modelerSystem_);
+
+    FillContext fillCtx(problemeHebdo->weekInTheYear * 168 + 0,
+                        problemeHebdo->weekInTheYear * 168 + 167);
     LinearProblemBuilder linearProblemBuilder(fillersCollection);
 
     if (solver == nullptr)
     {
-        linearProblemBuilder.build(*ortoolsProblem, LP_Data, fillCtx);
+        linearProblemBuilder.build(*ortoolsProblem, *problemeHebdo->linear_problem_data_, fillCtx);
```

Comment:
Please add the following comment
Note that the modeler is only called for the 1st simulation week, this limitation must be lifted later, when appropriate solvers (e.g with warm start) is integrated.